shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java

```diff
@@ -158,8 +160,18 @@ public void startInitialization(@NonNull Context applicationContext, @NonNull Se
 
     initStartTimestampMillis = SystemClock.uptimeMillis();
     flutterApplicationInfo = ApplicationInfoLoader.load(appContext);
-    VsyncWaiter.getInstance((WindowManager) appContext.getSystemService(Context.WINDOW_SERVICE))
-        .init();
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      final DisplayManager dm = appContext.getSystemService(DisplayManager.class);
+      final Display primaryDisplay = dm.getDisplay(Display.DEFAULT_DISPLAY);
+      VsyncWaiter.getInstance(primaryDisplay.getRefreshRate()).init();
+    } else {
+      VsyncWaiter.getInstance(
```

Comment:
Factor the duplicated `VsyncWaiter.getInstance().init()` calls out of the if/else blocks.